dask/dataframe/core.py

```diff
@@ -2173,11 +2173,64 @@ def std(
     ):
         axis = self._validate_axis(axis)
         _raise_if_object_series(self, "std")
+
         meta = self._meta_nonempty.std(axis=axis, skipna=skipna)
+        is_df_like = is_dataframe_like(self._meta_nonempty)
+        needs_time_conversion = False
+        numeric_dd = self
+
+        if PANDAS_GT_120:
+            from .io import from_pandas
+            from .numeric import to_numeric
+
+            def convert_to_numeric(s):
+                if skipna:
+                    return to_numeric(s.dropna())
+
+                # to_numeric(pd.NaT) => -9223372036854775808 is why we need to do this
+                return to_numeric(s).mask(s.isnull(), np.nan)
+
+            if is_df_like:
+                time_cols = self._meta_nonempty.select_dtypes(
+                    include="datetime"
+                ).columns
+                if len(time_cols) > 0:
+                    needs_time_conversion = True
+
+                    # Ensure all columns are correct type. Need to shallow copy since cols will be modified
+                    if axis == 0:
+                        numeric_dd = self[meta.index].copy()
+                    else:
+                        numeric_dd = self.copy()
+
+                    # If there's different types, just convert everything to NaNs for the time columns
+                    if axis == 1 and len(time_cols) != len(self.columns):
+                        # This is faster that converting each columns to numeric when it's not necessary
+                        numeric_dd[time_cols.tolist()] = from_pandas(
+                            pd.DataFrame(
+                                {col: pd.Series([np.nan]) for col in time_cols},
+                                index=self.index,
+                            ),
+                            npartitions=self.npartitions,
+                        )
+                        needs_time_conversion = False
+                    else:
+                        # Convert timedelta and datetime columns to integer types so we can use var
+                        for col in time_cols:
+                            numeric_dd[col] = convert_to_numeric(numeric_dd[col])
+            else:
```

Comment:
An aside, and others might disagree with me here: I find this deeply nested branching kind of tough to read. Specifically: what "if" block is this attached to? Obviously I can scroll up and down, and some editor support can help here, but to me it might help with legibility to add some small comments like
```python
      else:   # not is_df_like
```

Reply:
I'm wondering if this might actually be more legible if the conditionals were collapsed, so it would be:

```python
if PANDAS_GT_120 and is_df_like:
    ....
elif PANDAS_GT_120 and not is_df_like:
    ...